From: Kir Kolyshkin <kir@openvz.org>
Date: Wed, 21 Jul 2010 15:38:33 +0000 (+0400)
Subject: bash_completion: vzlist's stderr goes to /dev/null
X-Git-Tag: vzctl-3.0.24.1~1
X-Git-Url: http://git.openvz.org/?p=vzctl;a=commitdiff_plain;h=cea1f9509b08fe44e7410668241ed8a8734d7cd6;hp=540ac78700dba92ac145d7f69d67d344c880ba04

bash_completion: vzlist's stderr goes to /dev/null

vzlist can write some warnings to stderr. This is not
what we want to see while working with bash_completion.

Example of what could happen before applying this patch:
 vzctl create <tab>Invalid value for IOPRIO=88, skipped

Signed-off-by: Kir Kolyshkin <kir@openvz.org>
---

diff --git a/etc/bash_completion.d/vzctl.sh.in b/etc/bash_completion.d/vzctl.sh.in
index 59c996e..e68331d 100644
--- a/etc/bash_completion.d/vzctl.sh.in
+++ b/etc/bash_completion.d/vzctl.sh.in
@@ -11,24 +11,25 @@ _get_ves()
 	case $cmd in
 		create)
 			# create a new CTID, by increasing the last one
-			local veids=`@SBINDIR@/vzlist -H -a -octid | tail -1`
+			local veids=`@SBINDIR@/vzlist -H -a -octid \
+				2>/dev/null | tail -1`
 			[ -n "$veids" ] || veids=100
 			echo $((veids+1))
 			;;
 		start|mount|umount|destroy|delete)
 			# stopped CTs
-			@SBINDIR@/vzlist -H -S -octid
-			@SBINDIR@/vzlist -H -S -oname | $rm_empty
+			@SBINDIR@/vzlist -H -S -octid 2>/dev/null
+			@SBINDIR@/vzlist -H -S -oname 2>/dev/null | $rm_empty
 			;;
 		stop|enter|exec*)
 			# running CTs
-			@SBINDIR@/vzlist -H -octid
-			@SBINDIR@/vzlist -H -oname | $rm_empty
+			@SBINDIR@/vzlist -H -octid 2>/dev/null
+			@SBINDIR@/vzlist -H -oname 2>/dev/null | $rm_empty
 			;;
 		*)
 			# All CTs
-			@SBINDIR@/vzlist -H -a -octid
-			@SBINDIR@/vzlist -H -a -oname | $rm_empty
+			@SBINDIR@/vzlist -H -a -octid 2>/dev/null
+			@SBINDIR@/vzlist -H -a -oname 2>/dev/null | $rm_empty
 			;;
 	esac
 }
@@ -195,7 +196,7 @@ _vzctl()
 				fi
 				# NAME or ID ?
 				LIST_OPT=`echo $ve | awk '/[a-zA-Z]/ {print "-N"}'`
-				local ips="`@SBINDIR@/vzlist -H -o ip $LIST_OPT $ve | grep -vi -`"
+				local ips="`@SBINDIR@/vzlist -H -o ip $LIST_OPT $ve 2>/dev/null | grep -vi -`"
 				COMPREPLY=( $( compgen -W "$ips all" -- $cur ) )
 				;;
 			--netif_del)
